File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        l=[]
        dfs(l,root)
        return l[k-1]
def dfs(l,root):
    if root==None:
        return
    dfs(l,root.left)
    l.append(root.val)
    dfs(l,root.right)

File: Data Structures & Algorithms/balanced-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int max=0;
    public boolean isBalanced(TreeNode root) {
        height(root);
        return max<=1;
    }
    public int height(TreeNode node){
        if(node == null) return 0;
        int lh=height(node.left);
        int rh=height(node.right);

        max=Math.max(max,Math.abs(lh-rh));
        return Math.max(lh,rh)+1;
    }
}
